pkg/system/phase4_configuring.go

```diff
@@ -66,12 +66,20 @@ func (r *Reconciler) ReconcilePhaseConfiguring() error {
 	if err := r.ReconcileSystemSecrets(); err != nil {
 		return err
 	}
-	util.KubeCreateOptional(util.KubeObject(bundle.File_deploy_scc_endpoint_yaml).(*secv1.SecurityContextConstraints))
-	if err := r.ReconcileObject(r.DeploymentEndpoint, r.SetDesiredDeploymentEndpoint); err != nil {
-		return err
-	}
-	if err := r.ReconcileHPAEndpoint(); err != nil {
-		return err
+	// No endpoint creation is required for remote noobaa client
+	annotationValue, exists := util.GetAnnotationValue(r.NooBaa.GetAnnotations(), "remote-client-noobaa")
+	annotationBoolVal := true
+	if exists {
+		annotationBoolVal = strings.ToLower(annotationValue) == strconv.FormatBool(true)
+	}
+	if !exists || !annotationBoolVal {
```

Comment:
since you repeat this code more than once, mayb extract this to a function `isRemoteClientNoobaa`